src/StreamMaster.h

```diff
@@ -45,9 +45,13 @@ template <typename Streamer> class StreamMaster {
         WriterTask(std::move(file_writer_task)) {
 
     for (auto &d : Demuxers) {
-      Streamers.emplace(std::piecewise_construct,
-                        std::forward_as_tuple(d.topic()),
-                        std::forward_as_tuple(broker, d.topic(), options));
+      try {
+        Streamers.emplace(std::piecewise_construct,
+                          std::forward_as_tuple(d.topic()),
+                          std::forward_as_tuple(broker, d.topic(), options));
+      } catch (std::exception &e) {
+        LOG(Sev::Critical, "{}", e.what());
+      }
       Streamers[d.topic()].setSources(d.sources());
```

Comment:
Does this behave correctly if the code in `try` throws?

Reply:
The behaviour was in any case correct, nevertheless it is more clean as we discuss this morning.